com.unity.multiplayer.mlapi/Runtime/Interest/InterestNode.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI.Connection;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace MLAPI.Interest
+{
+    public abstract class InterestObjectStorage : ScriptableObject
+    {
+        public abstract void Query(in NetworkClient client, HashSet<NetworkObject> results);
+        public abstract void AddObject(NetworkObject obj);
+        public abstract void RemoveObject(NetworkObject obj);
```

Comment:
perhaps `InterestManager` could implement this interface instead of implementing its very own `HandleSpawn`/`HandleDespawn`?

Reply:
oh wait, `InterestObjectStorage` is not an interface. okay, so let's put these (suddenly common) methods into an interface and implement that interface on `InterestManager`, `InterestObjectStorage`, `InterestNode` etc.?

Reply:
I gave it a try with IInterestHandler for you to check out.  I have gone back and forth trying to unify InterestNode and InterestManager.  They basically do the same things as you've astutely observed.  If I did that I'm not sure I'd still need the interface.